javascript/preflight.js

```diff
@@ -0,0 +1,46 @@
+function runPreFlightChecks() {
+  const isHttp      = window.location.protocol == 'http:',
+        isLocalhost = window.location.hostname == 'localhost'
+        isUnsafe    = !(isHttp && isLocalhost);
+
+  if (isUnsafe) {
+    const isHttpMessage = [
+      'This page is running over the web over HTTP.',
+      'You should use HTTP and rotate your access token.',
+    ].join(' ');
+    alert(isHttpMessage);
+  }
+
+  // We will:
+  // - Make a request to the github rate limit endpoint
+  //   (This will not affect the rate limit)
+  // - Check what scopes we have access to
+  const token             = new FourthWall.getQueryVariables().token,
+        ghUrl             = 'https://api.github.com/rate_limit',
+        authGhUrl         = ghUrl + '?access_token=' + token;
+
+  fetch(authGhUrl)
+    .then(function (response) { return response.headers; })
+    .then(function (headers)  { return headers.get('x-oauth-scopes')  })
+    .then(function (scopes)   { return scopes.split(', '); })
+    .then(function (scopes)   {
+      const allowedScopes = ['public_repo', 'repo:status', 'repo_deployment'];
```

Comment:
even `public_repo` seems unnecessary for fourth wall ?

> Grants read/write access to code, commit statuses, collaborators, and deployment statuses for public repositories and organizations. Also required for starring public repositories.

Reply:
Good catch

Reply:
@tlwr did you validate it works without `public_repo` ? I'm not a fourth wall user so my comment was just a guess.

Reply:
yes, only those two scopes are needed